src/containers/Petitions.js

```diff
@@ -1,51 +1,97 @@
 import React from 'react';
 import Helmet from 'react-helmet';
 import { connect } from 'react-redux';
-import { fetchPetitions } from 'actions/PetitionsActions';
+import { withRouter } from 'react-router';
+import citySuggestionFormatter from 'helpers/citySuggestionFormatter';
+import { petitionsPath } from 'helpers/petitionUrls';
+import {
+  fetchPetitions,
+  fetchCity,
+  updateCityFilterValue
+} from 'actions/PetitionsActions';
 import settings from 'settings';
 import Petitions from 'components/Petitions';
 import getPetitions from 'selectors/petitions';
 
-const PetitionsContainer = React.createClass({
+const PetitionsContainer = withRouter(React.createClass({
   componentWillMount () {
     if (this.props.location.action === 'PUSH') {
       this.props.fetchPetitions(this.props);
+      this.props.fetchCity(this.props);
     }
   },
 
   componentWillReceiveProps (nextProps) {
-    if (nextProps.params.page !== this.props.params.page) {
+    if (this.props.location.pathname !== nextProps.location.pathname) {
       this.props.fetchPetitions(nextProps);
+      this.props.fetchCity(nextProps);
     }
   },
 
+  getAutocompleteProps: ({ router, updateCityFilterValue, cityFilterValue }) => ({
+    name: 'city-filter',
+    endpoint: 'cities',
+    suggestionFormatter: citySuggestionFormatter,
+    getFormValue: (suggestion) => suggestion,
+    suggestionsLimit: 4,
+    helper: {
+      value: { data: cityFilterValue },
+      onChange (newValue) {
+        updateCityFilterValue(newValue);
+
+        router.push(petitionsPath({
+          city: (newValue.id ? newValue : '')
+        }));
+      },
+      onBlur () {}
+    },
+    html: { placeholder: settings.petitionsPage.filters.city.placeholder }
+  }),
+
+  getTitle () {
+    return this.props.city
+      ? `${settings.petitionsPage.titleLocalized} ${this.props.city.name}`
+      : settings.petitionsPage.title;
+  },
+
   render () {
+    const title = this.getTitle();
```

Comment:
stuff like this always seems like a smell to me. Can't we know the `title` already when we render?